integration-tests/src/mock_network/setup.rs

```diff
@@ -0,0 +1,177 @@
+    (mock_network_actor, client_actor, view_client)
+}
+
+#[cfg(test)]
+mod test {
+    use crate::mock_network::setup::setup_mock_network;
+    use actix::{Actor, System};
+    use futures::{future, FutureExt};
+    use near_actix_test_utils::run_actix;
+    use near_chain_configs::Genesis;
+    use near_client::GetBlock;
+    use near_logger_utils::init_integration_logger;
+    use near_network::test_utils::{open_port, WaitOrTimeoutActor};
+    use nearcore::config::GenesisExt;
+    use nearcore::{load_test_config, start_with_config};
+
+    // Just a test to test that the basic mocknet setup works
+    // This test first starts a localnet with one validator node that generates 20 blocks
+    // to generate a chain history
+    // then start a mock network with this chain history and test that
+    // the client in the mock network can catch up these 20 blocks
+    #[test]
+    fn test_mocknet_basic() {
+        init_integration_logger();
+
+        // first set up a network with only one validator and generate some blocks
+        let genesis = Genesis::test(vec!["test0".parse().unwrap()], 1);
+        let mut near_config = load_test_config("test0", open_port(), genesis.clone());
+        near_config.client_config.min_num_peers = 0;
+
+        let dir = tempfile::Builder::new().prefix("test0").tempdir().unwrap();
+        let path1 = dir.path().clone();
+        run_actix(async move {
+            let nearcore::NearNode { view_client, .. } =
+                start_with_config(path1, near_config).expect("start_with_config");
+
+            WaitOrTimeoutActor::new(
+                Box::new(move |_ctx| {
+                    actix::spawn(view_client.send(GetBlock::latest()).then(|res| {
+                        let latest_height =
+                            if let Ok(Ok(block)) = res { block.header.height } else { 0 };
+                        if latest_height >= 20 {
+                            System::current().stop()
+                        }
+                        future::ready(())
+                    }));
+                }),
+                100,
+                60000,
+            )
+            .start();
+        });
+
+        // start the mock network to simulate a new node "test1" to sync up
+        let dir1 = tempfile::Builder::new().prefix("test1").tempdir().unwrap();
+        let mut near_config1 = load_test_config("test1", open_port(), genesis);
+        near_config1.client_config.min_num_peers = 1;
+        run_actix(async move {
+            let (_mock_network, _client, view_client) =
+                setup_mock_network(dir1.path().clone(), dir.path().clone(), &near_config1);
+            WaitOrTimeoutActor::new(
+                Box::new(move |_ctx| {
+                    actix::spawn(view_client.send(GetBlock::latest()).then(|res| {
+                        if let Ok(Ok(block)) = res {
+                            if block.header.height >= 20 {
+                                System::current().stop()
```

Comment:
could we maybe also check that block hash from here matches the one from line 142?